SourceryTests/Parsing/FileParser + AssociatedTypeSpec.swift

```diff
@@ -0,0 +1,114 @@
+                                }
+                            """
+                            let expectedProtocol = Protocol(name: "Foo")
+                            expectedProtocol.associatedTypes["Bar"] = AssociatedType(
+                                name: "Bar",
+                                typeName: TypeName("Codable")
+                            )
+                            expect(parse(code)).to(equal([expectedProtocol]))
+                        }
+                    }
+
+                    context("with associatedType constrained to a known type") {
+                        it("extracts associatedType properly") {
+                            let code = """
+                                protocol A {}
+                                protocol Foo {
+                                    associatedtype Bar: A
+                                }
+                            """
+                            let protocolA = Protocol(name: "A")
+                            let protocolFoo = Protocol(name: "Foo")
+                            protocolFoo.associatedTypes["Bar"] = AssociatedType(
+                                name: "Bar",
+                                typeName: TypeName("A"),
+                                type: protocolA
+                            )
+                            expect(parse(code)).to(equal([protocolA, protocolFoo]))
+                        }
+                    }
+
+                    context("with associatedType constrained to a composite type") {
+                        it("extracts associatedType properly") {
+                            let code = """
+                                protocol Foo {
+                                    associatedtype Bar: Encodable & Decodable
+                                }
+                            """
+                            let expectedType = ProtocolComposition(
+                                inheritedTypes: ["Encodable", "Decodable"],
+                                composedTypeNames: [TypeName("Encodable"), TypeName("Decodable")]
+                            )
+                            let expectedProtocol = Protocol(name: "Foo")
+                            expectedType.parent = expectedProtocol
+                            expectedProtocol.associatedTypes["Bar"] = AssociatedType(
+                                name: "Bar",
+                                typeName: TypeName("Encodable & Decodable"),
+                                type: expectedType
+                            )
+                            let actualProtocol = parse(code).first
+                            expect(actualProtocol).to(equal(expectedProtocol))
+                            let actualType = (actualProtocol as? SourceryProtocol)?.associatedTypes.first?.value.type
+                            expect(actualType).to(equal(expectedType))
+                        }
+                    }
+                }
+            }
+        }
+    }
+}
```

Comment:
Can we add a test case when using a typealias for associated type constraint?

Reply:
Thanks, adding this test led to an issue discovery with my code, fixed it too.